io.c

```diff
@@ -8490,11 +8517,31 @@ rb_p_result(int argc, const VALUE *argv)
 
 /*
  *  call-seq:
- *     p(obj)              -> obj
- *     p(obj1, obj2, ...)  -> [obj, ...]
- *     p()                 -> nil
+ *    p(object)   -> obj
+ *    p(*objects) -> array of objects
+ *    p           -> nil
+ *
+ *  For each object +obj+, executes:
+ *
+ *    $stdout.write(obj.inspect, "\n")
  *
- *  For each object, directly writes _obj_.+inspect+ followed by a
+ *  With one object given, returns the object;
+ *  with multiple objects given, returns an array containing the objects;
+ *  with no object given, returns +nil+.
+ *
+ *  Examples:
+ *
+ *    r = Range.new(0, 4)
+ *    p r                 # => 0..4
+ *    p [r, r, r]         # => [0..4, 0..4, 0..4]
+ *    p                   # => nil
+ *
+ *  Output:
+ *
+ *     0..4
+ *     [0..4, 0..4, 0..4]
+
+  directly writes _obj_.+inspect+ followed by a
```

Comment:
This looks wrong, it appears to be missing the starting `*` to signify it is a comment (it's technically still a comment, so it compiles fine, but it needs to be fixed before it is committed).

Reply:
All from 'directly' to the end is residue from the previous text, which should have been, and now is, deleted.